src/main/java/com/lambdaschool/foundation/models/HistoricalIncome.java

```diff
@@ -0,0 +1,96 @@
+package com.lambdaschool.foundation.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "historicalincome")
+public class HistoricalIncome extends Auditable
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long incid;
+
+    @NotNull
+    private int year;
+
+    @NotNull
+    private int individualincome;
+
+    @NotNull
+    private int householdincome;
+
+    @ManyToOne
+    @JoinColumn(name = "cityid")
+    @NotNull
+    @JsonIgnoreProperties(value = "historicalincome")
+    private City city;
+
+    public HistoricalIncome()
+    {
+    }
+
+    public HistoricalIncome(
+        @NotNull int year,
+        @NotNull int individualincome,
+        @NotNull int householdincome,
+        @NotNull City city)
+    {
+        this.year = year;
+        this.individualincome = individualincome;
+        this.householdincome = householdincome;
+        this.city = city;
+    }
+
+    public long getIncid()
+    {
+        return incid;
+    }
+
+    public void setIncid(long incid)
```

Comment:
Since we are generating the ID I think we should get rid of this setter as it's something we don't want to be changed.

Reply:
While we are generating the id, we still want to have the ability to change it if we want. Because the id is being generated locally on our backend, the ID isn't a parameter in any of the constructors. Being that the Id field is private we want to retain the ability to update it through the setter